src/ra_server.erl

```diff
@@ -827,17 +827,22 @@ handle_leader({transfer_leadership, Member},
            [LogId, Member]),
     {leader, State, [{reply, {error, unknown_member}}]};
 handle_leader({transfer_leadership, ServerId},
-              #{cfg := #cfg{log_id = LogId}} = State) ->
-    ?DEBUG("~ts: transfer leadership to ~w requested",
-           [LogId, ServerId]),
-    %% TODO find a timeout
-    gen_statem:cast(ServerId, try_become_leader),
-    {await_condition,
-     State#{condition =>
-            #{predicate_fun => fun transfer_leadership_condition/2,
-              timeout => #{effects => [],
-                           transition_to => leader}}},
-     [{reply, ok}]};
+              #{cfg := #cfg{log_id = LogId},
+                cluster := Cluster} = State) ->
+    case Cluster of
+        #{ServerId := #{voter_status := #{membership := Membership}}} when Membership =/= voter ->
+            ?DEBUG("~ts: transfer leadership requested but non-voter member ~w", [LogId, ServerId]),
+            {leader, State, [{reply, {error, non_voter_member}}]};
```

Comment:
make the error `{error, non_voter}` please

Reply:
Also please update the type spec fr the transfer_leadership function in the `ra` module and perhaps the docs also.

Reply:
I've updated the docs, but I think the function spec does not require any changes. The `{error, non_voter}` is already covered under `{error, term()}` in `ra:transfer_leadership/2` function specification."